torch/_library/custom_ops.py

```diff
@@ -502,6 +532,19 @@ def adinplaceorview_impl(keyset, *args, **kwargs):
                 with_keyset=True,
             )
 
+        if self._device_arg_id is not None:
+            device_arg_id: int = self._device_arg_id  # silent linter error
+
+            def backend_select(*args, **kwargs):
+                device = args[device_arg_id].type
+                if device not in self._backend_fns:
+                    raise RuntimeError(
+                        f"{self._name} does not have a kernel registered for {device}"
+                    )
+                return self._backend_fns[device](*args, **kwargs)
```

Comment:
for this to be faithful to how built-in PyTorch operators work, we should have with_keyset=True and have this do `opoverload.redispatch(keyset, *args, **kwargs)`